### PR TITLE
Fix #694: use Destination in CamelSources

### LINKED ISSUE #694: Camel Source usage of Desitination
The Camel Source should adpot `Destination` type, https://docs.google.com/document/d/1EAf9NLwDX3c9DS8lr6KxjtgRXUmkFGIcG6DeAYd5UHI/edit#

### PR DESCRIPTION
**Pull request by nicolaferraro**
Fixes #694

## Proposed Changes

  * Implement `Destination` in CamelSources

I'd like to merge this on master and 0.10.x.
I'll later upgrade `CamelSource` on to use the `pkg/controller` module in order to fully leverage the new URIResolver (e.g. to have immediate reaction when destination becomes available). That will be done on master only.

**Release Note**

```release-note
```

### CODE REVIEW

**Comment by nicolaferraro** [comment]
Is CI broken?

**Comment by lionelvillard** [comment]
Rebase and try again. See https://github.com/knative/eventing-contrib/pull/669#issuecomment-549490266

**Comment by nicolaferraro** [comment]
I see I'm already in sync with master.

/retest

**Comment by lionelvillard** [comment]
@matzew @slinkydeveloper The kafka performance test seems to be causing problems:

```
knative.dev/eventing-contrib/test/test_images/kafka_performance
knative.dev/eventing-contrib/test
/usr/local/go/pkg/tool/linux_amd64/link: exit status 1
# knative.dev/eventing-contrib/camel/source/cmd/controller
/usr/local/go/pkg/tool/linux_amd64/link: flushing $WORK/b677/exe/a.out: write $WORK/b677/exe/a.out: cannot allocate memory
/usr/local/go/pkg/tool/linux_amd64/link: exit status 2
# knative.dev/eventing-contrib/kafka/channel/cmd/channel_dispatcher
/tmp/go-build553557579/b592/_pkg_.a: not an object file
panic: runtime error: index out of range
```

**Comment by slinkydeveloper** [comment]
@lionelvillard I'll investigate, i really don't get what is about

**Comment by slinkydeveloper** [comment]
I just run a couple of times ` bash test/presubmit-tests.sh --build-tests` but it passes...

File: camel/source/pkg/apis/sources/v1alpha1/camelsource_types.go [hunk 1]
```diff
@@ -137,6 +138,22 @@ func (s *CamelSourceStatus) MarkSink(uri string) {
...
 
+// MarkSinkWarnDeprecated sets the condition that the source has a sink configured and warns ref is deprecated.
+func (s *CamelSourceStatus) MarkSinkWarnRefDeprecated(uri string) {
+	s.SinkURI = uri
+	if len(uri) > 0 {
+		c := apis.Condition{
+			Type:     CamelConditionSinkProvided,
+			Status:   corev1.ConditionTrue,
+			Severity: apis.ConditionSeverityError,
+			Message:  "Using deprecated object ref fields when specifying spec.sink. Update to spec.sink.ref. These will be removed in 0.11.",
```
Comment (lionelvillard) [inline camel/source/pkg/apis/sources/v1alpha1/camelsource_types.go]: this feels strange as the next release is 1.11. Maybe replace by "These will be removed in a future release"

Comment (nicolaferraro) [reply camel/source/pkg/apis/sources/v1alpha1/camelsource_types.go]: Right, I've fixed the message also in the other sources that were printing the same info.

**Comment by nicolaferraro** [comment]
It seems it's passing now.

**Pull request merged**